doc/MANUAL.txt

```diff
@@ -249,62 +310,85 @@ The process testr follows is:
    credentials.
 2. Execute testr run.
 3. testr queries for concurrency.
-4. testr will make a callout request to provision that many instances.
-   The provisioning callout needs to synchronise source code and do any other
-   per-instance setup at this stage.
-5. testr will make callouts to execute tests, supplying files that should be
+4. testr queries for available profiles.
+5. If testr needs to query test ids, it will provision one instance for each
+   profile, and use that instance to query test ids **per profile**. NB: until
+   testr gets an online scheduler, it will loop around the profiles one at a
+   time.
+6. testr will then provision as many instances as needed to execute the tests.
+   The provisioning callout is responsible for synchronising source
+   code and do any other per-instance preparation.
+7. testr will make callouts to execute tests, supplying files that should be
    copied into the execution environment. Note that instances may be used for
    more than one command execution.
-6. testr will callout to dispose of the instances after the test run completes.
+8. testr will callout to dispose of the instances after the instance is no
+   longer needed.
 
 Instances may be expensive to create and dispose of. testr does not perform
 any caching, but the callout pattern is intended to facilitate external
 caching - the provisioning callout can be used to pull environments out of
 a cache, and the dispose to just return it to the cache.
 
-Configuring environment support
--------------------------------
+Configuring profiles and environments
+-------------------------------------
 
-There are three callouts that testrepository depends on - configured in
-.testr.conf as usual. For instance::
+There are five callouts that testrepository depends on - configured in
+.testr.conf as usual.
 
-  instance_provision=foo -c $INSTANCE_COUNT
-  instance_dispose=bar $INSTANCE_IDS
-  instance_execute=quux $INSTANCE_ID $FILES -- $COMMAND
+For instance::
+
+  list_profiles=echo py27 py34
+  default_profiles=echo py27
+  instance_provision=foo -c $INSTANCE_COUNT $PROFILE
+  instance_dispose=bar $PROFILE $INSTANCE_IDS 
+  instance_execute=quux $PROFILE $INSTANCE_ID $FILES -- $COMMAND
 
 These should operate as follows:
 
+* list_profiles should return profile names. Profile names are simple text
```

Comment:
This is the only place I've seen a definition of how a profile can be named. Maybe that definition should be extracted elsewhere, highlighting the fact that they can't have whitespace or commas in them.